src/mbox/message-box.lisp

```diff
@@ -179,6 +198,23 @@ This function sets the result as `handler-result' in `item'. The return of this
               (bt2:condition-notify withreply-cvar)))
           (handler-fun)))))
 
+
+(declaim (ftype (function (message-box/bt)
+                          (values &optional))
+                ensure-thread-is-running))
+
+(defun ensure-thread-is-running (msgbox)
+  (with-slots (queue-thread thread-lock)
+      msgbox
+    (bt2:with-lock-held (thread-lock)
```

Comment:
I would tend to say this lock is not necessary, because `ensure-thread-is-running` is called from within `withreply-lock`.

Reply:
Locking/unlocking in this performance critical area takes additional time.